pydicom/dataset.py

```diff
@@ -599,6 +608,33 @@ def get_item(self, key):
             return self[key]
         return data_elem
 
+    def _dataset_slice(self, slice):
+        """Return a slice that has the same properties as the original
+        dataset. That includes properties related to endianess and VR handling,
+        and the specific character set. No element conversion is done, e.g.
+        elements of type RawDataElement are kept.
+        """
+        tags = self._slice_dataset(slice.start, slice.stop, slice.step)
+        dataset = Dataset({tag: self.get_item(tag) for tag in tags})
+        dataset.read_implicit_vr = self.read_implicit_vr
+        dataset.read_little_endian = self.read_little_endian
+        dataset.is_little_endian = self.is_little_endian
+        dataset.is_implicit_VR = self.is_implicit_VR
+        dataset.read_encoding = self.read_encoding
+        return dataset
+
+    def write_like_original(self):
```

Comment:
Property instead?

Reply:
I worry that the name might cause confusion with the `Dataset.save_as(fname, write_like_original)` parameter.

`Dataset.is_original_encoding`? Something similar?

Reply:
`is_original_encoding` sounds good - will change this. I generally struggle to find good names...

Reply:
Done.